stellar-contract-env-host/src/host.rs

```diff
@@ -389,6 +396,38 @@ impl Host {
         // Resolve the function symbol and invoke contract call
         vm.invoke_function_raw(self, SymbolStr::from(func).as_ref(), args)
     }
+
+    pub fn invoke_function(&mut self, hf: HostFunction, args: ScVec) -> Result<ScVal, HostError> {
+        let contract_id = match &args.as_vec()[0] {
+            ScVal::Object(Some(ob)) => self.to_host_obj(ob),
+            _ => Err(HostError::General("bad contract id")),
+        }?;
+
+        let symbol: Symbol = match &args.as_vec()[1] {
+            ScVal::Symbol(s) => s
+                .as_vec()
+                .as_slice()
+                .try_into()
```

Comment:
👀 I'll see if I can improve this type of thing in the generated xdr so we don't have this type of repetition pop up everywhere as the code base grows.

Reply:
You can drop the `.as_vec` at least for now.